stdlib/public/core/Integers.swift

```diff
@@ -1315,6 +1315,152 @@ extension BinaryInteger {
+    case 4:  return self == 10000 as UInt16
+    case 5:  return self == 100000 as UInt32
+    case 6:  return self == 1000000 as UInt32
+    case 7:  return self == 10000000 as UInt32
+    case 8:  return self == 100000000 as UInt32
+    case 9:  return self == 1000000000 as UInt32
+    case 10: return self == 10000000000 as UInt64
+    case 11: return self == 100000000000 as UInt64
+    case 12: return self == 1000000000000 as UInt64
+    case 13: return self == 10000000000000 as UInt64
+    case 14: return self == 100000000000000 as UInt64
+    case 15: return self == 1000000000000000 as UInt64
+    case 16: return self == 10000000000000000 as UInt64
+    case 17: return self == 100000000000000000 as UInt64
+    case 18: return self == 1000000000000000000 as UInt64
+    case 19: return self == 10000000000000000000 as UInt64
+    default:
+      // If this is 64-bit or less we can't have a higher power of 10
+      if self.bitWidth <= 64 { return false }
+
+      // Quickly check if parts of the bit pattern fits the power of 10.
+      //
+      // 10^0                                     1
+      // 10^1                                  1_01_0
+      // 10^2                               1_10_01_00
+      // 10^3                             111_11_01_000
+      // 10^4                          100111_00_01_0000
+      // 10^5                        11000011_01_01_00000
+      // 10^6                      1111010000_10_01_000000
+      // 10^7                   1001100010010_11_01_0000000
+      // 10^8                 101111101011110_00_01_00000000
+      // 10^9               11101110011010110_01_01_000000000
+      // 10^10           10010101000000101111_10_01_0000000000
+      // ...
+      // Column 1 is some "gibberish", which cannot be checked easily
+      // Column 2 is always the last two bits of the exponent
+      // Column 3 is always 01
+      // Column 4 is the trailing zeros, in equal number to the exponent value
+      //
+      // We check if Column 2 matches the last two bits of the exponent and
+      // Column 3 matches 0b01.
+      guard (self >> exponent)._lowWord & 0b1111 ==
+        ((exponent << 2) | 0b01) & 0b1111 else { return false }
+
+      // Now time for the slow path.
+      return self._slowIsPower(of: 10)
+    }
+  }
+
+  /// Returns `true` iff `self` is a power of the given `base`.
+  ///
+  /// This serves as the slow path for `isPower(of:)`; it is based on a generic
+  /// implementation that works for any input `base`.
+  @usableFromInline
+  internal func _slowIsPower(of base: Self) -> Bool {
+    // If self is 1 (i.e. any base to the zero power), return true.
+    if self == 1 { return true }
+
+    // Here if base is 0, 1 or -1, return true iff self equals base.
+    if base.magnitude <= 1 { return self == base }
+
+    // At this point, we have base.magnitude >= 2. Repeatedly perform
+    // multiplication by a factor of base, and check if it can equal self.
```

Comment:
The comment here doesn't match the implementation. Please update.

Reply:
Done!